src/Experiment/src/CustomAttributeBuilder.cs

```diff
@@ -0,0 +1,581 @@
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace System.Reflection.Emit.Experimental
+{
+    public class CustomAttributeBuilder
+    {
+        internal ConstructorInfo m_con;
+        private object?[] m_constructorArgs;
+        internal byte[] m_blob;
+
+        public ConstructorInfo Con { get => m_con;  }
+
+        // public constructor to form the custom attribute with constructor and constructor
+        // parameters.
+        public CustomAttributeBuilder(ConstructorInfo con, object?[] constructorArgs) :
+            this(con, constructorArgs, Array.Empty<PropertyInfo>(), Array.Empty<object>(), Array.Empty<FieldInfo>(), Array.Empty<object>())
+        {
+        }
+
+        // public constructor to form the custom attribute with constructor, constructor
+        // parameters and named properties.
+        public CustomAttributeBuilder(ConstructorInfo con, object?[] constructorArgs, PropertyInfo[] namedProperties, object?[] propertyValues) :
+            this(con, constructorArgs, namedProperties, propertyValues, Array.Empty<FieldInfo>(), Array.Empty<object>())
+        {
+        }
+
+        // public constructor to form the custom attribute with constructor and constructor
+        // parameters.
+        public CustomAttributeBuilder(ConstructorInfo con, object?[] constructorArgs, FieldInfo[] namedFields, object?[] fieldValues) :
+            this(con, constructorArgs, Array.Empty<PropertyInfo>(), Array.Empty<object>(), namedFields, fieldValues)
+        {
+        }
+
+        // public constructor to form the custom attribute with constructor and constructor
+        // parameters.
+        public CustomAttributeBuilder(ConstructorInfo con, object?[] constructorArgs, PropertyInfo[] namedProperties, object?[] propertyValues, FieldInfo[] namedFields, object?[] fieldValues)
+        {
+            ArgumentNullException.ThrowIfNull(con);
+            ArgumentNullException.ThrowIfNull(constructorArgs);
+            ArgumentNullException.ThrowIfNull(namedProperties);
+            ArgumentNullException.ThrowIfNull(propertyValues);
+            ArgumentNullException.ThrowIfNull(namedFields);
+            ArgumentNullException.ThrowIfNull(fieldValues);
+
+#pragma warning disable CA2208 // Instantiate argument exceptions correctly, combination of arguments used
+            if (namedProperties.Length != propertyValues.Length)
+                throw new ArgumentException("namedProperties, propertyValues differ");
+            if (namedFields.Length != fieldValues.Length)
+                throw new ArgumentException("namedFields, fieldValues differ");
+#pragma warning restore CA2208
+
+            if ((con.Attributes & MethodAttributes.Static) == MethodAttributes.Static ||
+                (con.Attributes & MethodAttributes.MemberAccessMask) == MethodAttributes.Private)
+                throw new ArgumentException("Bad Constructor");
```

Comment:
Can we provide better exception messages like: `The passed-in constructor is either static or private`